mlflow/models/signature.py

```diff
@@ -98,34 +105,45 @@ def from_dict(cls, signature_dict: Dict[str, Any]):
         inputs = Schema.from_json(signature_dict["inputs"])
         if "outputs" in signature_dict and signature_dict["outputs"] is not None:
             outputs = Schema.from_json(signature_dict["outputs"])
-            return cls(inputs, outputs)
         else:
-            return cls(inputs)
+            outputs = None
+        if (parameters := signature_dict.get("parameters")) is not None:
+            parameters = ParamSchema.from_json(parameters)
+            return cls(inputs, outputs, parameters)
+        else:
+            return cls(inputs, outputs)
 
     def __eq__(self, other) -> bool:
         return (
             isinstance(other, ModelSignature)
             and self.inputs == other.inputs
             and self.outputs == other.outputs
+            and self.parameters == other.parameters
         )
 
     def __repr__(self) -> str:
         return (
             "inputs: \n"
             "  {}\n"
             "outputs: \n"
-            "  {}\n".format(repr(self.inputs), repr(self.outputs))
+            "  {}\n"
+            "parameters: \n"
+            "  {}\n".format(repr(self.inputs), repr(self.outputs), repr(self.parameters))
         )
 
 
 def infer_signature(
-    model_input: Any, model_output: "MlflowInferableDataset" = None
+    model_input: Any,
+    model_output: "MlflowInferableDataset" = None,
+    parameters: Optional[Dict[str, Any]] = None,
```

Comment:
@serena-ruan @harupy @BenWilson2 Before merge / release, we should also make sure to mark all of the new `params` arguments in `infer_signature`, `predict()`, etc. as `@experimental`. We should also mark `ParamSpec` as `@experimental`.